lxd/storage/drivers/driver_pure_util.go

```diff
@@ -37,6 +38,51 @@ func (p *pureError) Error() string {
 	return strings.TrimSuffix(p.Errors[0].Message, ".")
 }
 
+// Matches returns true if the error status code is equal to the provided status code and the error message
+// contains the provided substring.
+func (p *pureError) Matches(statusCode int, substring string) bool {
```

Comment:
not enjoying that we need this hacky approach, but ill reserve judgement until i see where its needed.

Reply:
If you made substring a variadic arg you could use this function in `IsNotFoundError`

Reply:
Completely agree and am option to suggestions, but the main issue is that most errors (with just few rare exceptions) return status code `400`. Therefore, the substring matching is the important part of the function.

Reply:
> If you made substring a variadic arg you could use this function in IsNotFoundError

Yes indeed. Thanks.